File: patch_datepicker.js
const fs = require('fs');
const filepath = 'frontend/app/components/molecules/DatePicker/DatePicker.tsx';
let content = fs.readFileSync(filepath, 'utf8');

content = content.replace(
  `import { useTheme } from 'next-themes';`,
  `import { useTheme } from 'next-themes';\nimport { useFloating, offset, flip, shift, autoUpdate, FloatingPortal } from '@floating-ui/react';`
);

content = content.replace(
  `const [isOpen, setIsOpen] = useState(false);`,
  `const [isOpen, setIsOpen] = useState(false);\n  const { refs, floatingStyles } = useFloating({\n    open: isOpen,\n    onOpenChange: setIsOpen,\n    middleware: [offset(4), flip(), shift({ padding: 8 })],\n    whileElementsMounted: autoUpdate\n  });`
);

content = content.replace(
  `className={cn(
          commonStyles.inputWrapper,
          themeStyles.inputWrapper,`,
  `ref={refs.setReference}\n        className={cn(\n          commonStyles.inputWrapper,\n          themeStyles.inputWrapper,`
);

content = content.replace(
  `{isOpen && (
          <div
            id={calendarId}
            role="dialog"`,
  `{isOpen && (
          <FloatingPortal>
          <div
            ref={refs.setFloating}
            style={{ ...floatingStyles, zIndex: 1000 }}
            id={calendarId}
            role="dialog"`
);

content = content.replace(
  `            {renderCalendarFooter()}
          </div>
        )}`,
  `            {renderCalendarFooter()}
          </div>
          </FloatingPortal>
        )}`
);

fs.writeFileSync(filepath, content);
console.log("DatePicker patched");

File: patch_datepicker.py
import sys
path = 'frontend/app/components/molecules/DatePicker/DatePicker.tsx'
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

content = content.replace("import { useTheme } from 'next-themes';", "import { useTheme } from 'next-themes';\nimport { useFloating, offset, flip, shift, autoUpdate, FloatingPortal } from '@floating-ui/react';")
content = content.replace("const [isOpen, setIsOpen] = useState(false);", "const [isOpen, setIsOpen] = useState(false);\n  const { refs, floatingStyles } = useFloating({ open: isOpen, onOpenChange: setIsOpen, middleware: [offset(8), flip(), shift({ padding: 8 })], whileElementsMounted: autoUpdate });")
content = content.replace("ref={containerRef}", "ref={(node) => { containerRef.current = node; refs.setReference(node); }}")
content = content.replace("<div className={cn(commonStyles.calendarContainer, themeStyles.calendarContainer)}>", "<FloatingPortal>{isOpen && (<div ref={refs.setFloating} style={{ ...floatingStyles, zIndex: 1000 }} className={cn(commonStyles.calendarContainer, themeStyles.calendarContainer)}>")
content = content.replace("      )}", "      </div>)}</FloatingPortal>")
content = content.replace("  return (\n    <div\n      className={cn(", "  return (\n    <div\n      className={cn(") 

with open(path, 'w', encoding='utf-8') as f:
    f.write(content)
